### PR TITLE
Replace contextlib.ExitStack by direct cm invocations

### PR DESCRIPTION
**Pull request by keszybz**
I think it's nicer to keep things simple. Unfortunately black
turns the first chunk into a complete mess, so it needs to be disabled.
With python 3.9+ we will be able to drop the line continuations.

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -3676,15 +3678,14 @@ def insert_partition(
...
+    if args.root_partno == partno:
+        luks_format_root(args, loopdev, False, False, True)
+        cm = luks_setup_root(args, loopdev, False, True)
+    else:
+        cm = contextlib.nullcontext(partition(loopdev, partno))
 
-        path = dev if dev is not None else partition(loopdev, partno)
-        run(["dd", f"if={blob.name}", f"of={path}", "conv=nocreat,sparse"])
+    with cm as dev:
+        run(["dd", f"if={blob.name}", f"of={dev}", "conv=nocreat,sparse"])
```
Comment (DaanDeMeyer) [inline mkosi/__init__.py:3688]: This doesn't handle the case where `dev` is `None`

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: mkosi/__init__.py (hunk 1 above)
Comment (keszybz) [reply mkosi/__init__.py]: Ooops, indeed. Should be OK now.

**Comment by keszybz** [comment]
I restarted the CI here — it failed in arch/gpt_*, and I'm not sure if it was related to this PR or an unrelated transient issue.

**Comment by DaanDeMeyer** [comment]
Yeah there's something that got released recently causing timeouts in the Arch boot tests, but the current failures aren't related to this PR so LGTM

**Pull request merged**